bot/moderation/moderation.py

```diff
@@ -64,6 +64,54 @@ async def report_error(self, ctx: commands.Context, error: commands.CommandError
             await ctx.author.send(f"Ich konnte leider keinen Nutzer namens **{user}** finden. :confused:\nHast du dich "
                                   f"möglicherweise vertippt?")
 
+    @commands.group(name='purge', hidden=True)
+    @commands.has_role(constants.ROLE_ID_MODERATOR)
+    @command_log
+    async def purge_messages(self, ctx: commands.Context, channel: Optional[discord.TextChannel], amount: int):
+        """Command Handler for the `purge` command.
+
+        Allows moderators to delete the specified amount of messages in a channel. After invocation, a confirmation
+        message with all relevant information will be posted by SAM which the author needs to confirm in order to
+        proceed with the operation. If the moderator chooses an invalid amount of messages (negative or to high), a
+        temporary error message will be posted by the bot to inform the user.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+            channel (Optional[discord.TextChannel]): The channel in which the messages should be deleted.
+            amount (int): The amount of messages which need to be purged.
+        """
+        await ctx.message.delete()
```

Comment:
Why do you automatically delete the message? If it is in the channel to be deleted it will be deleted anyway, otherwise you would only want it to be deleted if it is a botonly channel right?

Otherwise we should do this with all channels right?

Reply:
Because if I want to delete 10 messages, then 10 messages should be deleted and not 9 + the invokation of the command.

Not really. I think its okay if people can see that a command has been invoked by someone. If we delete the initial message everytime, no one will really know who invoked a command. For other things like report, suggest, etc its generally a good idea to delete the invokation because you don't really want other people to see it or it doesn't make sense because there will be some sort of message posted anyway.